app/src/main/java/com/ojasvi/memify/ui/screens/PhotoReasoningScreen.kt

```diff
@@ -0,0 +1,244 @@
+) {
+
+    val photoReasoningUiState by viewModel.uiState.collectAsState()
+
+    val coroutineScope = rememberCoroutineScope()
+    val imageRequestBuilder = ImageRequest.Builder(LocalContext.current)
+    val imageLoader = ImageLoader.Builder(LocalContext.current).build()
+
+    PhotoReasoningScreen(
+        uiState = photoReasoningUiState,
+        onReasonClicked = { inputText, selectedImage ->
+            coroutineScope.launch {
+                val bitmap = try {
+                    val imageReq = imageRequestBuilder.data(selectedImage)
+                        .size(768)
+                        .precision(Precision.EXACT)
+                        .build()
+
+                    val result = imageLoader.execute(imageReq)
+                    if (result is SuccessResult) {
+                        (result.drawable as BitmapDrawable).bitmap
+                    } else {
+                        null
+                    }
+                } catch (e: Exception) {
+                    null
+                }
+
+                // Call reason function with inputText and bitmap
+                if (bitmap != null) {
+                    viewModel.reason(inputText, bitmap)
+                }
+            }
+        }
+    )
+}
+
+@Composable
+fun PhotoReasoningScreen(
+    uiState: PhotoReasoningUiState = PhotoReasoningUiState.Loading,
+    onReasonClicked: (String, Uri) -> Unit = { _, _ -> }
+) {
+    var userQuestion by rememberSaveable { mutableStateOf("") }
+    var imageUri by rememberSaveable { mutableStateOf<Uri?>(null) }
+
+    val pickMedia = rememberLauncherForActivityResult(
+        ActivityResultContracts.PickVisualMedia()
+    ) { selectedImageUri ->
+        selectedImageUri?.let {
+            imageUri = selectedImageUri
+        }
+    }
+
+    Column(
+        modifier = Modifier
+            .padding(all = 16.dp)
+            .verticalScroll(rememberScrollState())
+    ) {
+        AsyncImage(
+            model = imageUri,
+            contentDescription = null,
+            modifier = Modifier
+                .padding(4.dp)
+                .fillMaxWidth(0.5f).align(Alignment.CenterHorizontally)
+        )
+        Card(
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            Row(
+                modifier = Modifier.padding(top = 16.dp)
+            ) {
+                IconButton(
+                    onClick = {
+                        pickMedia.launch(
+                            PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly)
+                        )
+                    },
+                    modifier = Modifier
+                        .padding(all = 4.dp)
+                        .align(Alignment.CenterVertically)
+                ) {
+                    Icon(
+                        Icons.Rounded.Add,
+                        contentDescription = stringResource(R.string.add_image),
+                    )
+                }
```

Comment:
The modifier and clickable can be applied directly to the Icon. `IconButton` is not required.

Reply:
Done